tasktiger/migrations.py

```diff
@@ -0,0 +1,18 @@
+def migrate_executions_count(tiger):
+    """
+    Backfills ``t:task:<uuid>:executions_count`` by counting
```

Comment:
We'll mention this in the changelog. Should be sufficient, right?

Reply:
this probably warrants a `0.17` release + a big bold "**Breaking Change**" headline

Reply:
Yes, I was planning a major version bump. Also, I agree about making sure that the users are aware so that nothing works in an unexpected way for them.

That being said, this PR doesn't actually introduce any breaking changes, it just adds the counter.
The actual breaking changes will be introduced in a subsequent PR.

Also, if someone misses this info and they upgrade directly to the change from the next PR, the worst that would happen is that the retry counter would reset for all tasks started on a worker before this change.

Reply:
> the worst that would happen is that the retry counter would reset for all tasks started on a worker running on a version from before this change

Just pointing out that depending on the use case this could be very problematic. It's not outlandish to have a task where the execution/retry logic needs to be in sync with some sort of retryable business logic that cares about how many times it should attempt something before giving up/falling back to permanent failure logic.

Reply:
I'm not saying it couldn't be problematic and I'm definitely not saying we should ignore this. It's obviously a breaking change.

I was just pointing out (for no practical reason, I guess) that there's a good chance that the users who don't read the changelog before upgrading will not even encounter any issues.